cuqi/experimental/mcmc/_cwmh.py

```diff
@@ -0,0 +1,221 @@
+import numpy as np
+import cuqi
+from cuqi.experimental.mcmc import ProposalBasedSamplerNew
+from cuqi.array import CUQIarray
+from numbers import Number
+
+class CWMHNew(ProposalBasedSamplerNew):
+    """Component-wise Metropolis Hastings sampler.
+
+    Allows sampling of a target distribution by a component-wise random-walk
+    sampling of a proposal distribution along with an accept/reject step.
+
+    Parameters
+    ----------
+
+    target : `cuqi.distribution.Distribution` or lambda function
+        The target distribution to sample. Custom logpdfs are supported by using
+        a :class:`cuqi.distribution.UserDefinedDistribution`.
+    
+    proposal : `cuqi.distribution.Distribution` or callable method
+        The proposal to sample from. If a callable method it should provide a
+        single independent sample from proposal distribution. Defaults to a
+        Gaussian proposal.  *Optional*.
+
+    scale : float or ndarray
+        Scale parameter used to define correlation between previous and proposed
+        sample in random-walk.  *Optional*. If float, the same scale is used for
+        all dimensions. If ndarray, a (possibly) different scale is used for
+        each dimension.
+
+    initial_point : ndarray
+        Initial parameters. *Optional*
+
+    callback : callable, *Optional*
+        If set this function will be called after every sample.
+        The signature of the callback function is
+        `callback(sample, sample_index)`, where `sample` is the current sample
+        and `sample_index` is the index of the sample.
+        An example is shown in demos/demo31_callback.py.
```

Comment:
Is the way callbacks are illustrated in demo31 still relevant for the new samplers? Maybe add an issue to update the demo once the samplers have replaced the old samplers.

Reply:
good point, I recorded this in #378